build fails on sprite-file-dummy and data-source-custom

6>data.lib(sprite-file-dummy.obj) : error LNK2019: unresolved external symbol "public: virtual __thiscall SpriteFile::~SpriteFile(void)" (??1SpriteFile@@UAE@XZ) referenced in function "public: virtual void * __thiscall SpriteFile::`scalar deleting destructor'(unsigned int)" (??_GSpriteFile@@UAEPAXI@Z)
6>C:\foo8\freeserf\build\src\Debug\FreeSerf.exe : fatal error LNK1120: 1 unresolved externals
6>Done building project "FreeSerf.vcxproj" -- FAILED.

I can bypass by commenting this line out of src/CMakeLists.txt:
#list(APPEND DATA_SOURCES sprite-file-dummy.cc)

...but then another error shows up:
6>data.lib(data-source-custom.obj) : error LNK2019: unresolved external symbol "public: __thiscall SpriteFile::SpriteFile(void)" (??0SpriteFile@@QAE@XZ) referenced in function "public: __thiscall std::_Ref_count_obj<class SpriteFile>::_Ref_count_obj<class SpriteFile><>(void)" (??$?0$$V@?$_Ref_count_obj@VSpriteFile@@@std@@QAE@XZ)
6>data.lib(data-source-custom.obj) : error LNK2019: unresolved external symbol "public: bool __thiscall SpriteFile::load(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &)" (?load@SpriteFile@@QAE_NABV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@@Z) referenced in function "public: virtual class std::tuple<class std::shared_ptr<class Sprite>,class std::shared_ptr<class Sprite> > __thiscall DataSourceCustom::get_sprite_parts(enum Data::Resource,unsigned int)" (?get_sprite_parts@DataSourceCustom@@UAE?AV?$tuple@V?$shared_ptr@VSprite@@@std@@V12@@std@@W4Resource@Data@@I@Z)
6>C:\foo8\freeserf\build2\src\Debug\FreeSerf.exe : fatal error LNK1120: 2 unresolved externals
6>Done building project "FreeSerf.vcxproj" -- FAILED.

which can be avoided by removing the lines in src/CMakeLIsts.txt:

data-source-custom.h
data-source-custom.cc

.. and also commenting out these lines in removing the lines from src/data.cc:

//  sources_factories.push_back([](const std::string &path)->PDataSource{
//    return std::make_shared<DataSourceCustom>(path); });

The game compiles and runs normally with the above workarounds.  However, I'm interested in trying out custom datasources so I'd like to get this working.

--- a/src/sprite-file-sdl.cc
+++ b/src/sprite-file-sdl.cc
@@ -27,9 +27,6 @@ SpriteFile::SpriteFile() {
   IMG_Init(IMG_INIT_JPG | IMG_INIT_PNG | IMG_INIT_TIF);
 }
 
-SpriteFile::~SpriteFile() {
-}
-
 bool
 SpriteFile::load(const std::string &path) {
   SDL_Surface *image = IMG_Load(path.c_str());

--- a/src/sprite-file.h
+++ b/src/sprite-file.h
@@ -30,7 +30,7 @@
 class SpriteFile : public Sprite {
  public:
   SpriteFile();
-  virtual ~SpriteFile();
+  virtual ~SpriteFile() {}
 
   bool load(const std::string &path);
 

